Exposed API calls to both authentication login from Burp API

ganesh2183 opened the pull request:
Added api call to recorded login script feature.

ganesh2183 committed:
Update _burp_commander.py

Replaced code changes for application_login api call 'UsernameAndPasswordLogin' with 'RecordedLogin'.

ganesh2183 committed:
Merge pull request #1 from ganesh2183/recorded-login-api

Update _burp_commander.py

ganesh2183 committed:
Update _burp_commander.py

ganesh2183 committed:
Merge pull request #2 from ganesh2183/recorded-login-api-2

Update _burp_commander.py

ganesh2183 committed:
Expose API call for recorded login sequence

ganesh2183 committed:
Update _burp_commander.py

ganesh2183 committed:
Update _burpa.py

ganesh2183 committed:
Update _burp_commander.py

ganesh2183 committed:
Update _burpa.py

tristanlatr commented on burpa/_burp_commander.py:
```diff
@@ -120,22 +138,31 @@ def get_scan_configurations(names: Optional[Iterable[str]], json_strings: Option
+                            username=username, password=password,
+                            exclude_rules=exclude_rules, scan_configurations=scan_configurations)	
+                   
+            if label and script:
```
```suggestion
            elif label and script:
```

tristanlatr commented on burpa/_burpa.py:
```diff
@@ -166,20 +167,22 @@ def _start_scan(self, *targets: str, excluded: str = "", config: str = "", confi
                             app_user=app_user,
                             app_pass=app_pass,
+                            recorded_login_label=recorded_login_label,
+                            recorded_login_script=recorded_login_script,
```
You need to read the `recorded_login_script` file before passing it to the `_start_scan` function. 
To read a file one can use the `open` function.

ganesh2183 replied:
I have added the below lines in _burpa.py after line:170
#Parse recorded login script file(s)
     script_files = parse_commas_separated_str(recorded_login_script)
     script_files_content -[]
     for f in script_files:
          script_files_content.append(open(f,'r', encoding='utf').read())

Also in line:213 replaced script=script_files_content
label=recorded_login_label, script=script_files_content

But when i add json file in command it is adding extra double quotes on starting & end of the recorded script.

Ex: burpa scan https://example.com --recorded_login_label=IBXLogin --recorded_login_script=sample.json

"script":"[**"**{\"name\": \"Burp Suite Navigation Recorder\", \"url\": \"https://example.com/login\",\"isframe\": false}**"**]"

why and how to fix this issue??

ganesh2183 replied:
@tristanlatr Did you checked this?

tristanlatr reviewed (commented):
You're close to make it work, don't give up :)

ganesh2183 committed:
Update _burp_commander.py

ganesh2183 committed:
Update _burpa.py

ganesh2183 committed:
Update _burp_commander.py

ganesh2183 committed:
Update _burpa.py

ganesh2183 committed:
Update _burp_commander.py

ganesh2183 committed:
Update _utils.py

ganesh2183 committed:
Update _burpa.py

ganesh2183 commented:
Added API calls to recorded login script authentication login method, it works as expected. 

Updated code changes in utils.py, burpa.py and burp_commander.py files.

Could you please verify from your end and confirm on this.

burpa scan https://example.com/login --recorded_login_label=Login --recorded_login_script="Login.json"

ganesh2183 committed:
Update _burpa.py

ganesh2183 commented:
> Added API calls to recorded login script authentication login method, it works as expected.
> 
> Updated code changes in utils.py, burpa.py and burp_commander.py files.
> 
> Could you please verify from your end and confirm on this.
> 
> burpa scan https://example.com/login --recorded_login_label=Login --recorded_login_script="Login.json"

Please check this and confirm.

tristanlatr commented:
Have you tried it on your end ?
Because it seems that it's not even compiling.

ganesh2183 commented:
> Have you tried it on your end ? Because it seems that it's not even compiling.
I checked it is working for me.

tristanlatr commented:
> I checked it is working for me.

It cannot possibly be true, there is a syntax error :/

ganesh2183 commented:
> > I checked it is working for me.
> 
> It cannot possibly be true, there is a syntax error :/

What syntax error you are getting?
